src/nvim/normal.c

```diff
@@ -2939,6 +2939,9 @@ void check_visual_highlight(void)
  */
 void end_visual_mode(void)
 {
+  if (exiting) {
+    return;
+  }
```

Comment:
@justinmk  I've described how we have reached this in the commit message. I wasn't quite sure about this patch in general, let me know if there is a better way to do this.